CI build issue with Python version

After updating the package spec to remove Python `3.7` support (#144), the build action for publishing the package fails with the following error:
```log
##[debug]Evaluating condition for step: 'build wheel and source'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: build wheel and source
##[debug]Loading inputs
##[debug]Loading env
Run poetry build
##[debug]/usr/bin/bash -e /home/runner/work/_temp/[2](https://github.com/noaa-ocs-modeling/CoupledModelDriver/actions/runs/3578060447/jobs/6018376698#step:6:2)fc27ccd-00b7-47[3](https://github.com/noaa-ocs-modeling/CoupledModelDriver/actions/runs/3578060447/jobs/6018376698#step:6:3)d-a1de-8fd036527253.sh
The currently activated Python version 3.10.6 is not supported by the project (>=3.8,<3.10).
Trying to find and use a compatible version. 

  NoCompatiblePythonVersionFound

  Poetry was unable to find a compatible version. If you have one, you can explicitly use it via the "env use" command.

  at ~/.local/lib/python3.10/site-packages/poetry/utils/env.py:736 in create_venv
       732│                     python_minor = ".".join(python_patch.split(".")[:2])
       733│                     break
       73[4](https://github.com/noaa-ocs-modeling/CoupledModelDriver/actions/runs/3578060447/jobs/6018376698#step:6:4)│ 
       73[5](https://github.com/noaa-ocs-modeling/CoupledModelDriver/actions/runs/3578060447/jobs/6018376698#step:6:5)│             if not executable:
    →  73[6](https://github.com/noaa-ocs-modeling/CoupledModelDriver/actions/runs/3578060447/jobs/6018376698#step:6:6)│                 raise NoCompatiblePythonVersionFound(
       [7](https://github.com/noaa-ocs-modeling/CoupledModelDriver/actions/runs/3578060447/jobs/6018376698#step:6:7)37│                     self._poetry.package.python_versions
       73[8](https://github.com/noaa-ocs-modeling/CoupledModelDriver/actions/runs/3578060447/jobs/6018376698#step:6:8)│                 )
       73[9](https://github.com/noaa-ocs-modeling/CoupledModelDriver/actions/runs/3578060447/jobs/6018376698#step:6:9)│ 
       740│         if root_venv:
Error: Process completed with exit code 1.
##[debug]Finishing: build wheel and source
```

The link to the failing action (on v1.6.1) is:
> https://github.com/noaa-ocs-modeling/CoupledModelDriver/actions/runs/3578060447

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -41,5 +41,7 @@ jobs:
         with:
           name: build
           path: dist
+      - name: install Poetry
+        uses: abatilo/actions-poetry@v2.2.0          
       - name: upload wheel and source
         run: poetry publish --username __token__ --password ${{ secrets.PYPI_TOKEN }}

